backend/src/nodes/impl/color_transfer/linear_histogram.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+import numpy as np
+
+from nodes.impl.image_utils import to_uint8
+
+__author__ = "Daniel Steinberg"
+__copyright__ = "Copyright 2022, Daniel Steinberg"
+__credits__ = ["Daniel Steinberg"]
+__license__ = "MIT"
+__version__ = "1.0.0"
+__maintainer__ = "Daniel Steinberg"
+__link__ = "https://github.com/dstein64/colortrans"
+
+
+def matrix_sqrt(X):
+    eig_val, eig_vec = np.linalg.eig(X)
+    return eig_vec.dot(np.diag(np.sqrt(eig_val))).dot(eig_vec.T)
+
+
+def linear_histogram_transfer(
+    img: np.ndarray,
+    ref_img: np.ndarray,
+) -> np.ndarray:
+    """
+    Transfers the color distribution from the source to the target image
+    using the Linear Histogram Matching..
+
+    This implementation is based on to the Hertzmann, Aaron. "Algorithms
+    for Rendering in Artistic Styles." Ph.D., New York University, 2001.
+    """
+
+    shape = img.shape
+
+    # Convert HxWxC image to a (H*W)xC matrix.
+    content = to_uint8(img).reshape(-1, shape[-1])
```

Comment:
Question: do we really need to convert our images to uint8 here just to convert them back later? I just don't see why we wouldn't get the same result without it?

Reply:
My bad. I added this code before knowing the img.dtype was float: I then forgot to optimize this.
It works without the conversion.
Waiting for your feedback on the following conversation before pushing